tools/hsmtool.c

```diff
@@ -595,31 +596,38 @@ static int dumponchaindescriptors(const char *hsm_secret_path, const char *old_p
 	} while (bip32_key_from_seed(bip32_seed, sizeof(bip32_seed),
 				     version, 0, &master_extkey) != WALLY_OK);
 
-	if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PUBLIC, &enc_xpub) != WALLY_OK)
-		errx(ERROR_LIBWALLY, "Can't encode xpub");
+	if (show_secrets) {
+		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PRIVATE,
+					&enc_xkey) != WALLY_OK)
+			errx(ERROR_LIBWALLY, "Can't encode xpriv");
+	} else {
+		if (bip32_key_to_base58(&master_extkey, BIP32_FLAG_KEY_PUBLIC,
+					&enc_xkey) != WALLY_OK)
+			errx(ERROR_LIBWALLY, "Can't encode xpub");
+	}
```

Comment:
Why we do not dump the xpub too when the `show_secrets`, IMHO there is no meaning to run two command to know pub and private.

It is also true that from the secret we know the xpub too, but looks like strange to me required more that one step to have a full dump?

There is any reason to have this private **or** public logic that I am missing?

Reply:
I think the motivation for this command is have those descriptors that you can load into an external wallet. Therefore you either need the xpriv if you want a full wallet or the xpub if you want a watch-only wallet; you don't need both.

Furthermore, if we were to dump the three xprivs and also three xpubs, the output gets a bit messy and confusing.

Reply:
Make sense